File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> list = new ArrayList<>();
        Queue<Node> q = new LinkedList<>();
        
        for(int i=0; i<progresses.length; i++) q.add(new Node(progresses[i], speeds[i]));
        
        while(!q.isEmpty()) {
            int n = q.size();
            int cnt = 0;
            for(int i=0; i<n; i++) {
                Node cur = q.poll();
                q.add(new Node(cur.p + cur.s, cur.s));
            }
            
            while(!q.isEmpty()) {
                if(q.peek().p >= 100) {
                    cnt++;
                    q.poll();
                } else {
                    break;
                }
            }
            if(cnt > 0) list.add(cnt);
        }
        int[] answer = new int[list.size()];
        for(int i=0; i<answer.length; i++) {
            answer[i] = list.get(i);
        }
        return answer;
    }
    
    class Node {
        int p, s;
        public Node(int p, int s) {
            this.p = p;
            this.s = s;
        }
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    days = []

    # 각 기능이 배포되기까지 남은 날 수 계산
    for i in range(len(progresses)):
        day = (100 - progresses[i]) // speeds[i]
        if (100 - progresses[i]) % speeds[i] != 0:
            day += 1
        days.append(day)
    
    # 배포 가능한 기능의 개수를 세어줌
    max_day = days[0]
    count = 0
    
    for day in days:
        if day <= max_day:
            count += 1
        else:
            answer.append(count)
            count = 1
            max_day = day
    
    answer.append(count)
    
    return answer